toolbox

```diff
@@ -1580,6 +1580,14 @@ run()
         fi
     fi
 
+    if ! $podman_command exec \
+                 --user "$USER" \
+                 "$toolbox_container" \
+                 sh -c "test -d $PWD" >/dev/null 2>&3; then
+        echo "Directory $PWD does not exist in container $toolbox_container, try cd && !!" >&2
```

Comment:
I wish we could avoid another potentially slow `podman exec` (do the `exec`, invoke `sh`, etc.) before the *main* `podman exec` that offers the interactive shell. Most people running `toolbox enter` or `toolbox run` are hopefully not going to be hitting this error, so it makes sense to optimize for the non-failure case.

How about doing it after the *main* `podman exec` fails with an exit code of 127? That way the extra `podman exec` will only be run when things have already failed.

(I wish we could also somehow avoid the other `podman exec` that's already there to do the fallback-to-bash, but that one seems more difficult to get rid off because it involves the binary that gets invoked.)

Reply:
I took the liberty to adjust it as above, and I see that @HarryMichal already ported your change to his Go version.